### PR TITLE
Adding Content for SQL PaaS Solution

### PR DESCRIPTION
**Pull request by v-sabiraj**
Publisherid, offerid and solution id needs to be changed

### CODE REVIEW

**Comment by v-spadarthi** [comment]
@v-rucdu :Please review.

**Review by v-dvedak — DISMISSED** [review]

**Comment by v-laanjana** [comment]
@v-sabiraj could you please check if we can merge this PR?

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-ExecutionTimeAnomaly.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                        , 0),
+      AdditionalInfo = additional_information_s, timeSlice = floor(TimeGenerated, timeSliceSize));
+  processedData
+  //Bin the data into 1h windows, taking the average of exeuction time
+  | summarize round(avg(DurationMs), 2), min(TimeGenerated), max(TimeGenerated) by PrincipalName, ApplicationName, bin(TimeGenerated, timeSliceSize)
+  //Summarise by user and application and create list ready for anomaly detection
+  | summarize make_list(avg_DurationMs), make_list(min_TimeGenerated), make_list(max_TimeGenerated) by PrincipalName, ApplicationName
+  | extend series_decompose_anomalies(list_avg_DurationMs, scoreThreshold, -1, 'linefit')
+  | mv-expand TimeAnomaly=series_decompose_anomalies_list_avg_DurationMs_ad_flag, WindowStart=list_min_TimeGenerated, WindowEnd=list_max_TimeGenerated
+  | project WindowStart, WindowEnd, PrincipalName, ApplicationName, TimeAnomaly
```
Comment (shainw) [inline Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-ExecutionTimeAnomaly.yaml:41]: ResoureId is dropped in this line and the query will fail later on the last project statement as ResourceId is not brought through the left side of the join.

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-VolumeAffectedRowsStatefulAnomalyOnDatabase.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                  Error = case( additional_information_s has 'error_code', toint(extract("<error_code>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+)", 1, additional_information_s))
+                        , 0),
+                  State = case( additional_information_s has 'error_state', toint(extract("<error_state>([0-9.]+)", 1, additional_information_s))
+                        , additional_information_s has 'failure_reason', toint(extract("<failure_reason>Err ([0-9.]+), Level ([0-9.]+)", 2, additional_information_s))
+                        , 0)
+        | extend QuantityColumn = column_ifexists(monitoredColumn, 0)
+        | sort by TimeGenerated desc
+        | extend RowNumber = row_number(),
+                             WindowType = case( TimeGenerated >= ago(detectionWindow), 'detection',
```
Comment (shainw) [inline Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-VolumeAffectedRowsStatefulAnomalyOnDatabase.yaml:43]: detectionWindow doesn't appear to be a valid column or variable and is causing the query to fail.

**Review by shainw — CHANGES_REQUESTED** [review]
Some fixes needed in the hunting queries.  Additionally, in general, the Analytic Rules are duplicated with just some different parameters passed thru.  It looks like you have 2 categories of queries using the same base query. Each category would be best to be combined into a single, [custom details](https://docs.microsoft.com/en-us/azure/sentinel/customize-alert-details), this way only 1 query is need for each of these sets of queries -  Detection-Errors* and Detection-Hotwords*

**Comment by v-spadarthi** [comment]
@v-sabiraj : Please resolve the comments and conflicts as well.

**Comment by v-spadarthi** [comment]
@v-sabiraj : Please resolve the comments and conflicts as well.

**Comment by v-spadarthi** [comment]
@v-sabiraj : Please resolve the comments and conflicts as well.

**Comment by v-spadarthi** [comment]
@v-sabiraj : Please resolve the comments and conflicts as well.

**Comment by v-spadarthi** [comment]
@v-sabiraj : Please resolve the comments and conflicts as well.

**Comment by v-amolpatil** [comment]
@v-sabiraj  : Please resolve the comments and conflicts as well.

**Comment by v-sabiraj** [comment]
Resolved conflicts, can you please approve and merge this as the content is already live in content hub and we are just adding it in GitHub, thanks.

**Comment by v-spadarthi** [comment]
@v-sabiraj : As discussed, please have a look and provide an update. Thanks!

**Comment by v-spadarthi** [comment]
@v-sabiraj : As discussed, please have a look and provide an update. Thanks!

**Comment by v-sabiraj** [comment]
Hi @shainw, actually these analytic rules and hunting queries are already live in content hub and we are just adding in GitHub. Can you please help us on fixing these queries as we are not able to test in our environment, thanks.

**Comment by v-spadarthi** [comment]
Hi @shainw, actually these analytic rules and hunting queries are already live in content hub and we are just adding in GitHub. Can you please help us on fixing these queries as we are not able to test in our environment, thanks.

**Comment by v-sabiraj** [comment]
@shainw, done this change, can you please approve, thanks.

**Comment by NikTripathi** [comment]
@shainw Can you please provide your approval? Thanks.

**Comment by v-sabiraj** [comment]
@shainw Can you please provide your approval? Thanks.

**Comment by v-rucdu** [comment]
@shainw : Need your assistance here... please take a look... Thanks!!!

**Comment by v-rucdu** [comment]
@v-sabiraj : As discussed, please close the open comment from Shain and request him re-approval... Thanks!!

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-ExecutionTimeAnomaly.yaml (hunk 1 above)
Comment (shainw) [reply Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-ExecutionTimeAnomaly.yaml]: This is not resolved, The project statement on line 11 doesn't have ResourceId and so it fails later.

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-VolumeAffectedRowsStatefulAnomalyOnDatabase.yaml (hunk 2 above)
Comment (shainw) [reply Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-VolumeAffectedRowsStatefulAnomalyOnDatabase.yaml:43]: This also needs to be fixed still.

**Comment by shainw** [comment]
> Some fixes needed in the hunting queries. Additionally, in general, the Analytic Rules are duplicated with just some different parameters passed thru. It looks like you have 2 categories of queries using the same base query. Each category would be best to be combined into a single, [custom details](https://docs.microsoft.com/en-us/azure/sentinel/customize-alert-details), this way only 1 query is need for each of these sets of queries - Detection-Errors* and Detection-Hotwords*

The items I commented on before still need to be fixed.

**Comment by v-sabiraj** [comment]
Hi @shainw, I updated the queries, thanks.

**Review by shainw — APPROVED** [review]

**Pull request merged**